### PR TITLE
Add JSdoc comments for api wrapped functions

### PR DESCRIPTION
**Pull request by qur786**

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -292,18 +304,70 @@ function generateOptionsString(input?: GetINCHIOptions | GetINCHIExOptions): str
...
+ * Input: takes two argument
+ *  input - source InChI string ,
+ *  strict (optional) - if false, just briefly check for proper layout (prefix, version, etc.)
+ * Output: Returns code -1 , 0 , 1 , 2 , 3 , 4
+ * -1: InChI is valid and non-standard
+ *  0: InChI is valid and standard
+ *  1: InChI has invalid prefix
+ *  2: InChI has invalid version number (not equal to 1)
+ *  3: InChI has invalid layout
+ *  4: Checking InChI through InChI2InChI either failed or produced a result which does not match the source InChI string
```
Comment (maneetgoyal) [inline src/api.ts:333]: Study my last commit and update accordingly. 

Upon hovering over the function name, nicely formatted text should appear.

Example:

Screenshot 2021-08-31 at 4 02 35 PM

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

File: src/api.ts [hunk 2]
```diff
@@ -292,18 +304,94 @@ function generateOptionsString(input?: GetINCHIOptions | GetINCHIExOptions): str
...
+ * @returns One amongst -1 , 0 , 1 , 2 , 3 , 4
+ * - -1: InChI is valid and non-standard
+ * - 0: InChI is valid and standard
+ * - 1: InChI has invalid prefix
+ * - 2: InChI has invalid version number (not equal to 1)
+ * - 3: InChI has invalid layout
+ * - 4: Checking InChI through InChI2InChI either failed or produced a result which does not match the source InChI string
+ * @example
+ * ```ts
+ * const status = CheckINCHI("InChI=1S/C3H5NO/c1-2-5-3-4-1/h3H,1-2H2/p+1",false);
```
Comment (maneetgoyal) [inline src/api.ts:340]:
```suggestion
 * const status = CheckINCHI("InChI=1S/C3H5NO/c1-2-5-3-4-1/h3H,1-2H2/p+1", false);
```

File: src/api.ts [hunk 3]
```diff
@@ -292,18 +304,94 @@ function generateOptionsString(input?: GetINCHIOptions | GetINCHIExOptions): str
...
+ * console.log(status);
+ * //0
```
Comment (maneetgoyal) [inline src/api.ts:342]:
```suggestion
 * // 0
```

File: src/api.ts [hunk 4]
```diff
@@ -292,18 +304,94 @@ function generateOptionsString(input?: GetINCHIOptions | GetINCHIExOptions): str
...
 
+/**
+ * Returns length of the string
+ * @param {string} input An Inchi string
+ * @returns {number} length of the string
+ * @example
+ * ```ts
+ * const length = GetStringLength("VNWKTOKETHGBQD-UHFFFAOYSA-N");
+ * console.log(length);
+ * //27
```
Comment (maneetgoyal) [inline src/api.ts:357]:
```suggestion
 * // 27
```

File: src/api.ts [hunk 5]
```diff
@@ -292,18 +304,94 @@ function generateOptionsString(input?: GetINCHIOptions | GetINCHIExOptions): str
...
+ * const output = GetStructFromINCHI("InChI=1S/C2H6O/c1-2-3/h3H,2H2,1H3");
+ * console.log(output.status);
+ * //0
+ * console.log(output.data.atom.x);
+ * //0
+ * console.log(output.data.atom.y);
+ * //0
+ * console.log(output.data.atom.z);
+ * //0
+ * ```
```
Comment (maneetgoyal) [inline src/api.ts:393]: Handle spacing properly. (Same as previous comments)

File: src/api.ts [hunk 6]
```diff
@@ -292,18 +304,94 @@ function generateOptionsString(input?: GetINCHIOptions | GetINCHIExOptions): str
...
+ *  - data - data contains structure created from input (Inchi) string
+ *   -- atom - Array of atom objects which contains various data like atom co-ordinates (x,y,z) , neighbor, bondType , bondStereo, elName, numBonds, numIsoh, isotopicMass, radical, charge
+ *   -- stereo0D - array of num_stereo0D 0D stereo elements or NULL
+ *   -- numAtoms - number of atoms in the structure
+ *   -- numStereo0D - number of 0D stereo elements
+ *   -- szMessage - A string containing Error/warning ASCIIZ message
+ *   -- szLog - log-file ASCIIZ string, contains a human-readable list of recognized options and possibly an Error/warn message
+ *   -- warningFlags - A 2d Array conataining warnings , [x][y]:
+ *            -- x=0 => Reconnected if present in InChI otherwise Disconnected/Normal
+ *            -- x=1 => Disconnected layer if Reconn. layer is present
```
Comment (maneetgoyal) [inline src/api.ts:379]: Please post a snapshot of how it looks upon hover.

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

File: src/api.ts (hunk 6 above)
Comment (qur786) [reply src/api.ts:379]: image

Comment (qur786) [reply src/api.ts:379]: image

File: src/api.ts [hunk 7]
```diff
@@ -292,18 +304,94 @@ function generateOptionsString(input?: GetINCHIOptions | GetINCHIExOptions): str
...
```
Comment (maneetgoyal) [inline src/api.ts:379]: You can completely remove this comment. JSDoc style comment are processed by the TypeDoc library. TypeDoc is already aware of the various type definition so it'll auto-hyperlink the output type.

Screenshot 2021-08-31 at 6 34 01 PM

Focus more on **function description** and **example usage**. Rest should be auto-handled by TypeDoc.

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

**Review by maneetgoyal — APPROVED** [review]

**Pull request merged**